pyccel/codegen/printing/ccode.py

```diff
@@ -1022,10 +1022,18 @@ def function_signature(self, expr, print_arg_names = True):
         if not args:
             arg_code = 'void'
         else:
-            arg_code = ', '.join('{}'.format(self.function_signature(i.var, False))
-                        if isinstance(i.var, FunctionAddress)
-                        else '{0}'.format(self.get_declare_type(i.var)) + (i.name if print_arg_names else '')
-                        for i in args)
+            def get_var_arg(arg, var):
+                code = "const " if var.is_const else ""
+                code += self.get_declare_type(var)
+
+                if print_arg_names:
+                    code += arg.name
+                return code
```

Comment:
Just an opinion, this could be compactly written like this
```python3
qual = "const " * var.is_const
name = arg.name * print_arg_var
return qual + self.get_declare_type(var) + name
```